pages/docs/security.mdx

```diff
@@ -0,0 +1,47 @@
+# Security
+
+Security is a primary consideration when moving systems into production.  In this section we'll dive into how Inngest handles security, including endpoint security, encryption, standard practices, and how to add SAML authentication to your account.
+
+## Signing keys and SDK security
+
+Firstly, it's important to understand that in production all communication between Inngest and your servers is encrypted via TLS. The SDK also actively mitigates attacks by use of the signing key, a secret pre-shared key unique to each environment. By default, the signing key adds the following:
+- **Authentication:** requests to your endpoint are authenticated, ensuring that they originate from Inngest. Without knowledge of the signing key, requests are immediately rejected.
+- **Replay attack prevention:** requests are signed with a timestamp embedded, and old requests are rejected, even if the requests are authenticated correctly.
+
+**It's important that the signing key is kept secret.** If your signing key is exposed, it puts the security of your endpoints at risk. Note that it's possible to rotate signing keys with zero downtime.
+
+### Function registration + handshake
+
+Within Inngest, you write and serve functions on your own infrastructure using our SDKs. In order to run your functions, they must be synced, or registered, with your Inngest account. Without this, we won't know which functions you're serving or how to run them. Syncing functions is done via a secure handshake. Here's how the handshake works:
+1. After your SDK's endpoint is live, a PUT request to the endpoint initiates a secure handshake to the Inngest servers.
+2. The SDK sends function configuration to Inngest's API, with the signing key as a bearer token.
+3. The SDK idempotently updates your apps and functions. If there are no changes, nothing happens.
+
+This process is necessary for several reasons, largely as serverless environments are the lowest common denominator. Serverless environments have no default bootup/init process, which means serverless environments can't self-register. Secondly, serverless platforms such as AWS Lambda and Vercel create unique URLs for each function deployed, which can't be known ahead of time. The incoming PUT request allows the SDK to introspect the request's URL, which is then used for all function calls.
+
+Note that because the SDK reaches out to inngest.com to register, it never leaks the signing key to clients attempting registration, keeping your key secure.
```

Comment:
sync, "requests" vs. "reaches out" for more exact tone.
```suggestion
Note that because the SDK only sends HTTP requests to `api.inngest.com` to complete the sync, it never leaks the signing key to clients attempting registration, keeping your key secure.
```